### PR TITLE
docs: fix #2698

### LINKED ISSUE #2698: Update docs
There have been a number of StackOverflow and Discord posts regarding deprecated options in the docs, particularly cluster execution. Just wanted to start the issue to identify places in the docs that need updating so we can tackle them in a relatively organized way. If you find any, please reply here with a link to docs and what is outdated.

1. [`--cluster` in the tutorial](https://snakemake.readthedocs.io/en/stable/tutorial/additional_features.html#cluster-execution)

### PR DESCRIPTION
**Pull request by cmeesters**
### Description

"Just" a documentation update:

- deleted obsolete `--cluster` references in the tutorial section
- rephrased MPI section

References to SLURM Executor Plugin and snakemake-profiles in both changed paragraphs.

@johanneskoester today, I am not particularly good with words, but I hope its ok-ish.

### CODE REVIEW

**Comment by cmeesters** [comment]
PS this PR addresses issue #2698

PPS I just hoped to get my way with the semantic commits ...

**Comment by cmeesters** [comment]
At #2698 it has been commented that the API docs are missing, too. Not sure: Have these been auto-generated in the past @johanneskoester ?

**Comment by johanneskoester** [comment]
Yes, they are auto-generated. They are also there, but need some polishing: https://snakemake-api.readthedocs.io/en/stable/

And a link from the main docs as well.

File: docs/snakefiles/rules.rst [hunk 1]
```diff
@@ -2689,7 +2689,10 @@ Template rendering rules are always executed locally, without submission to clus
 MPI support
 -----------
 
-Highly parallel programs may use the MPI (:ref: message passing interface<https://en.wikipedia.org/wiki/Message_Passing_Interface>) to enable a program to span work across an individual compute node's boundary.
+Highly parallel programs may use the MPI (:ref: [message passing interface](https://en.wikipedia.org/wiki/Message_Passing_Interface)) to enable a program to span work across an individual compute node's boundary. 
+To actually use an HPC cluster with Snakemake, an [executor plugin is provided for the SLURM batch system](https://github.com/snakemake/snakemake-executor-plugin-slurm). You can find its documentation [here](https://github.com/snakemake/snakemake-executor-plugin-slurm/blob/main/docs/further.md).
+Users of different batch systems ought to refer to the [Snakemake profiles project](https://github.com/Snakemake-Profiles) for available plugins.
```
Comment (johanneskoester) [inline docs/snakefiles/rules.rst:2694]:
```suggestion
To actually use an HPC cluster with Snakemake, an [executor plugin is provided for the SLURM batch system](https://github.com/snakemake/snakemake-executor-plugin-slurm). You can find its documentation [here](https://github.com/snakemake/snakemake-executor-plugin-slurm/blob/main/docs/further.md).
Users of different batch systems are encouraged to [provide further plugins](https://snakemake.github.io/snakemake-plugin-catalog/#contributing) and/or share their Snakemake configuration via the [Snakemake profiles project](https://github.com/Snakemake-Profiles) project.
```

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**